### PR TITLE
[hotfix][connector-v2] fix pulsar source exceptions

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
1. fix the exception that pulsar discovering splits
2. fix exactly-once exception

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java [hunk 1]
```diff
@@ -127,6 +129,19 @@ public void open() {
 
     @Override
     public void run() throws Exception {
+        if (partitionDiscoveryIntervalMs > 0) {
+            new ScheduledThreadPoolExecutor(1, runnable -> {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java]: How to close this thread pool when terminates a task?

Comment (ashulin) [reply seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java]: Sorry, I forgot it.

**Review by EricJoy2048 — DISMISSED** [review]
LGTM

File: seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java [hunk 2]
```diff
@@ -127,6 +130,20 @@ public void open() {
...
     @Override
     public void run() throws Exception {
+        if (partitionDiscoveryIntervalMs > 0) {
+            executor = new ScheduledThreadPoolExecutor(1, runnable -> {
+                Thread thread = new Thread(runnable);
+                thread.setDaemon(true);
+                thread.setName("pulsar-split-discovery-executor");
+                return thread;
+            });
+            executor.schedule(this::discoverySplits, partitionDiscoveryIntervalMs, TimeUnit.MILLISECONDS);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java]: Use `try catch` wrap `this::discoverySplits` and log throwable message?

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java (hunk 2 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/enumerator/PulsarSplitEnumerator.java]: Submits a one-shot task? 

```java

executor.scheduleAtFixedRate(new Runnable() {
                    @Override
                    public void run() {
                        try {
                            this.discoverySplits();
                        } catch (Throwable e) {
                            log.error(..., e);
                        }
                    }
                }, 0, partitionDiscoveryIntervalMs, TimeUnit.MILLISECONDS);
```

**Review by EricJoy2048 — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Comment by hailin0** [comment]
LGTM

**Pull request merged**